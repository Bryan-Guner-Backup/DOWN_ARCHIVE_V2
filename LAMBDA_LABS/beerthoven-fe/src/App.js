import React from "react";
import { Route, Switch } from "react-router-dom";
import "./App.css";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { Layout, Menu } from "antd";
import { Security, SecureRoute, LoginCallback } from "@okta/okta-react";
import oktaConfig from "./okta-config";
import OktaApolloProvider from "./components/utils/OktaApolloProvider";
import LogoutButton from "./components/LogoutButton";
import Dashboard from "./components/pages/Dashboard";
import Login from "./components/pages/Login";
import People from "./components/pages/People";
import Event from "./components/pages/Event";
import Venue from "./components/pages/Venue";
import User from "./components/pages/User";
const { Sider, Content } = Layout;

function App() {
  return (
    <div className="App" data-testid="app">
      <Router>
        <Security {...oktaConfig}>
          <OktaApolloProvider>
            <Layout>
              <Sider
                style={{
                  overflow: "auto",
                  height: "100vh",
                  position: "fixed",
                  left: 0,
                }}
              >
                <Menu theme="dark">
                  <Menu.ItemGroup key="h1" title="Manage">
                    <Menu.Item key="1">
                      <Link to="/">Dashboard</Link>
                    </Menu.Item>
                    <Menu.Item key="2">
                      <Link to="/people">People</Link>
                    </Menu.Item>
                    <Menu.Item key="3">
                      <Link to="/event">Events</Link>
                    </Menu.Item>
                    <Menu.Item key="4">
                      <Link to="/venue">Venues</Link>
                    </Menu.Item>
                    <Menu.Item key="5">
                      <Link to="/user">Users</Link>
                    </Menu.Item>
                  </Menu.ItemGroup>
                </Menu>
                <div
                  style={{
                    paddingTop: "30px",
                    display: "grid",
                    placeItems: "center",
                  }}
                >
                  <LogoutButton />
                </div>
              </Sider>
              <Layout
                style={{
                  marginLeft: 200,
                  height: "100vh",
                }}
              >
                <Content style={{ margin: "24px 16px 0", overflow: "initial" }}>
                  <Switch>
                    <SecureRoute path="/login" component={Login} />
                    <SecureRoute path="/people" component={People} />
                    <SecureRoute path="/event" component={Event} />
                    <SecureRoute path="/venue" component={Venue} />
                    <SecureRoute path="/user" component={User} />
                    <Route
                      path={oktaConfig.callbackPath}
                      component={LoginCallback}
                    />
                    <SecureRoute exact path="/" component={Dashboard} />
                  </Switch>
                </Content>
              </Layout>
            </Layout>
          </OktaApolloProvider>
        </Security>
      </Router>
    </div>
  );
}

export default App;
