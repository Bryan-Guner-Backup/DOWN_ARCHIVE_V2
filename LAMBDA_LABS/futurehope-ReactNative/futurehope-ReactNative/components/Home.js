import React from 'react';
import { connect } from 'react-redux';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { NativeRouter, Route, Link } from "react-router-native"
import Icon from "react-native-vector-icons/FontAwesome";
import { widthPercentageToDP, heightPercentageToDP } from "../utils/PercenatageFix";
import { Footer, FooterTab, Button } from 'native-base';
import CalendarView from './calendar/CalendarView';
import SwitchNavigator from '../navigation/SwitchNavigator'

class Home extends React.Component{
    render() {
    return (
        <View style={styles.container}>
            <Text style={styles.text}>Welcome {this.props.user.fullName} </Text>
            <SwitchNavigator />
        </View>
    )
    }
}
const mapStateToProps = state => {
	return {
		user: state.user
	}
}

const styles = StyleSheet.create({
    text: {
        fontSize: 28,
        textAlign: 'center',
        maxWidth: ('60%'),
        fontWeight: 'bold',
        margin: ('15%'),
        color: 'white' 
    },
    container: {
        height: heightPercentageToDP('82.5%'),
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'orange' 
    }
});


export default connect(mapStateToProps)(Home);
