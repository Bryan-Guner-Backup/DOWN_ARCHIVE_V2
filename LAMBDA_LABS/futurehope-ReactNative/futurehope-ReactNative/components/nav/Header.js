import React from 'react';
import { View, Text, StyleSheet, ShadowPropTypesIOS } from 'react-native';
import { NativeRouter, Route, Link, useHistory } from "react-router-native"
import Icon from 'react-native-vector-icons/FontAwesome'
import { widthPercentageToDP, heightPercentageToDP } from '../../utils/PercenatageFix'

const Header = () => {
    const history = useHistory()
    function backHandler() {
        history.goBack()
    }

    return (
        <View style={styles.header}>
            <Icon style={styles.icon} name='arrow-left' onPress={backHandler} size={20} color='#ff9800' />
            <Text style={styles.text}>School in the Sky</Text>
            <Link style={styles.icon} to='/'><Icon name='home' color='#ff9800' size={20} /></Link>
        </View>
    )
};

const styles = StyleSheet.create({
    header: {
        flexDirection: 'row',
        height: heightPercentageToDP('9%'),
        paddingTop: heightPercentageToDP('6%'),
        paddingBottom: heightPercentageToDP('2%'),
        alignItems: 'center',
        borderBottomColor: '#eee',
        borderBottomWidth: 1,
        justifyContent: 'space-between'
    },
    text: {
        fontWeight: 'bold',
        fontSize: 18,
        textAlign: 'center',
        color: 'gray'
    },
    icon: {
        marginLeft: widthPercentageToDP('4%'),
        marginRight: widthPercentageToDP('4%')
    }
});

export default Header;
