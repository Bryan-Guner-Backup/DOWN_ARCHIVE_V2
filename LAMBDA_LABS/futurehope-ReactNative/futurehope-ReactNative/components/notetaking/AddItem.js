import React, { useState } from 'react';
import { View, Text, StyleSheet, TextInput, TouchableOpacity, KeyboardAvoidingView, } from 'react-native';
import { widthPercentageToDP, heightPercentageToDP } from '../../utils/PercenatageFix';

const AddItem = ({ title, addItem, text, onTextChange, handleModal }) => {
    const submitHandler = () => {
        addItem(text)
        handleModal()
    }

    return (
        <View style={styles.container}>
            <KeyboardAvoidingView behavior='height' keyboardVerticalOffset={80}>
                <View style={styles.container}>
                    <Text style={styles.text}>New Note</Text>
                </View>
                <TextInput multiline numberOfLines={10} onChangeText={text => onChangeText(text)} placeholder="Add item..." style={styles.input} onChangeText={onTextChange} value={text} />
            </KeyboardAvoidingView>
            <View style={styles.buttonsdesign}>
                <TouchableOpacity style={styles.button}  >
                    <Text onPress={handleModal} style={styles.buttonText}>Cancel </Text>
                    <Text onPress={submitHandler} style={styles.buttonText}>Add Note </Text>
                </TouchableOpacity>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: heightPercentageToDP('1%'),
        marginBottom: heightPercentageToDP('1%'),
    },
    input: {
        height: heightPercentageToDP('70%'),
        width: widthPercentageToDP('95%'),
        margin: widthPercentageToDP('2%'),
        fontSize: 16,
        borderColor: '#eee',
        borderWidth: widthPercentageToDP('.5%'),
        borderRadius: 5,
        textAlignVertical: 'top',
        padding: heightPercentageToDP('1%')

    },
    text: {
        fontSize: 24,
        textAlign: 'center',
        color: 'orange',
    },
    button: {
        backgroundColor: 'white',
        padding: widthPercentageToDP('2%'),
        margin: widthPercentageToDP('1%'),
        width: widthPercentageToDP('99%'),
        flexDirection: "row",
        justifyContent: "space-between",
    },
    buttonText: {
        color: 'white',
        fontSize: 20,
        textAlign: 'center',
        backgroundColor: '#FFB23D',
        padding: widthPercentageToDP('2%'),
        width: widthPercentageToDP('45%')
    }
});

export default AddItem;