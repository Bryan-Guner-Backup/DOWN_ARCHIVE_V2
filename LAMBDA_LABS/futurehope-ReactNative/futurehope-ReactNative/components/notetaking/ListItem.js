import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';

import { Link, useHistory } from "react-router-native";

import Icon from 'react-native-vector-icons/FontAwesome';
import { widthPercentageToDP } from '../../utils/PercenatageFix';

const ListItem = ({ item, deleteItem }) => {
    const history = useHistory()

    function onPress() {
        history.push(`/notetaking/noteedit/${item.id}`)
    }
    return (
        <TouchableOpacity onPress={onPress} style={styles.listItem}>
            <View style={styles.listItemView}>
                <Text style={styles.listItemText}>{item.title}</Text>
                <Icon color='#ff9800' size={20} name='chevron-right' />
            </View>
        </TouchableOpacity>

    )
}


const styles = StyleSheet.create({
    listItem: {
        padding: 15,
        backgroundColor: '#fff',
        borderBottomWidth: 1,
        borderColor: '#eee',
        borderWidth: 1
    },
    listItemView: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    listItemText: {
        fontSize: 18,
        width: widthPercentageToDP('83%'),
        maxWidth: 360
    }
});


export default ListItem;
