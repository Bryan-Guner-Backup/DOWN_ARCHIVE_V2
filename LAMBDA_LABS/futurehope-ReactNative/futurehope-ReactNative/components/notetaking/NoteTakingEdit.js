import React, { useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  KeyboardAvoidingView
} from "react-native";
import { useParams, useHistory } from "react-router-native";
import {
  widthPercentageToDP,
  heightPercentageToDP
} from "../../utils/PercenatageFix";
import Icon from "react-native-vector-icons/FontAwesome";

const EditItem = ({ items, editItem, edit, editChange, deleteItem }) => {
  const { id } = useParams();
  const history = useHistory()
  const [target, setTarget] = useState({});
  const [editNote, setEditNote] = useState(false);
  React.useEffect(() => {
    items.forEach(e => {
      if (e.id === id) {
        setTarget(e);
        editChange(target.content)
      }
    });
  }, [items, editNote]);

  const deletePress = () => {
    deleteItem(id)
    history.push("/notetaking/noteview")
  }

  return (
    <View style={styles.container}>
      <KeyboardAvoidingView behavior='height' keyboardVerticalOffset={50}>
        <View style={{ flexDirection: 'row', justifyContent: "space-between", alignItems: 'center', maxWidth: widthPercentageToDP('90%') }}>
          <Text style={styles.title}>{target.title}</Text>
          <Icon name='trash' color='grey' size={25} onPress={deletePress} />
        </View>
        {!target.content ? (
          <Text style={editNote ? styles.invis : styles.placeHolder}>
            Add note here...
          </Text>
        ) : (
            <Text style={editNote ? styles.invis : styles.contents}>
              {target.content}
            </Text>
          )}
        <TextInput
          placeholder="Add note here..."
          style={editNote ? styles.input : styles.invis}
          onChangeText={editChange}
          value={edit}
        />
      </KeyboardAvoidingView>
      <TouchableOpacity
        style={editNote ? styles.button : styles.invis}
        onPress={() => {
          editItem(edit, target.id);
          setEditNote(false);
        }}
      >
        {/* <Icon name='check' color='white' size= {20}/> */}
        <Text style={styles.buttonText}>Done</Text>
      </TouchableOpacity>

      <TouchableOpacity
        style={editNote ? styles.invis : styles.editContentBtn}
        onPress={() => setEditNote(true)}
      >
        <Text style={styles.contentBtnText}>Edit Note</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: heightPercentageToDP("82%"),
    textAlign: "left",
    marginLeft: widthPercentageToDP('5%'),
    marginRight: widthPercentageToDP('2%'),
    maxWidth: ('90%'),
  },
  invis: {
    display: "none"
  },
  top: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between"
  },
  title2: {
    fontSize: 30
  },
  title: {
    fontSize: widthPercentageToDP('10%'),
    fontWeight: "bold",
    padding: widthPercentageToDP('1%'),
    marginTop: heightPercentageToDP('1%')
  },
  edit: {
    color: "#FFB23D",
    textAlign: "right",
    fontSize: 20,
    fontWeight: "bold"
  },
  placeHolder: {
    marginLeft: widthPercentageToDP('.5%'),
    color: "grey"
  },
  input: {
    height: heightPercentageToDP('60%'),
    textAlignVertical: 'top',
    fontSize: 16,
    borderColor: '#eee',
    borderWidth: 1,
    padding: widthPercentageToDP('1%'),
    borderRadius: 5
  },
  button: {
    width: "100%",
    backgroundColor: "#FFB23D",
    position: "absolute",
    bottom: heightPercentageToDP('1%')
  },
  buttonText: {
    color: "white",
    textAlign: "center",
    fontWeight: "bold",
    padding: widthPercentageToDP('2%'),
    fontSize: 30
  },
  contents: {
    padding: widthPercentageToDP('1%'),
    color: "grey",
    fontSize: 20
  },
  editContentBtn: {
    width: "100%",
    backgroundColor: "#FFB23D",
    position: "absolute",
    bottom: heightPercentageToDP('1%')
  },
  contentBtnText: {
    color: "white",
    textAlign: "center",
    fontWeight: "bold",
    padding: widthPercentageToDP('2%'),
    fontSize: 20,
  },
  touchableOpacity:{
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
  }
});

export default EditItem;
