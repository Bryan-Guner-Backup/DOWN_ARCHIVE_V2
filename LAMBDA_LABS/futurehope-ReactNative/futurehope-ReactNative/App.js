import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, FlatList, Alert, AsyncStorage, TouchableOpacity } from 'react-native';
import Header from './components/nav/Header';
import Home from './components/Home'
import CalendarRoutes from './components/calendar/CalendarRoutes';
import { heightPercentageToDP } from "./utils/PercenatageFix";
import { NativeRouter, Route, Link, Switch, useHistory } from "react-router-native"
import NoteTakingRoutes from './components/notetaking/NoteTakingRoutes';
import BottomFooter from './components/nav/BottomFooter';

import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import thunkMiddleware from 'redux-thunk'
import reducer from './reducers'

import {decode, encode} from 'base-64'

if (!global.btoa) {  global.btoa = encode }
if (!global.atob) { global.atob = decode }

const middleware = applyMiddleware(thunkMiddleware)
const store = createStore(reducer, middleware)

const App = () => {


  return (
    <Provider store={store}>
      <NativeRouter>
        <View style={styles.container}>
          <Header />
          <Switch>
            <Route exact path='/' component={Home} />
            <Route path='/calendar'><CalendarRoutes /></Route>
            <Route path='/notetaking'><NoteTakingRoutes /></Route>
          </Switch>
          <BottomFooter style={styles.styledfooter} />
        </View>
      </NativeRouter>
    </Provider>
  )
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  styledfooter: {
    position: "absolute",
    bottom: heightPercentageToDP('0%')
  }
});


export default App;
