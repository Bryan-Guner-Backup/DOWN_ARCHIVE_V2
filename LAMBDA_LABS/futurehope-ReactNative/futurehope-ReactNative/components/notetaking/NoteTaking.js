import React, { useState } from 'react';
import { View, Text, StyleSheet, TouchableOpacity, FlatList, AsyncStorage, Modal, KeyboardAvoidingView } from 'react-native';
import ListItem from './ListItem';
import AddItem from './AddItem';
import Icon from "react-native-vector-icons/Entypo"
import { widthPercentageToDP, heightPercentageToDP } from "../../utils/PercenatageFix";
import { NativeRouter, Route, Link } from "react-router-native";


const NoteTaking = ({ items, deleteItem, addItem, text, onTextChange }) => {
    const [open, setOpen] = useState(false);


    const handleModal = () => setOpen(!open);

    return (
        <View style={styles.container}>

            <View style={styles.flatview}>
                <FlatList
                    data={items}
                    renderItem={({ item }) => <ListItem item={item} deleteItem={deleteItem} />}
                />
            </View>
            <Icon style={styles.plus} onPress={handleModal} color='#ff9800' size={60} name='circle-with-plus' />

            <Modal
                animationType="fade"
                visible={open}
                onDismiss={handleModal}>
                <AddItem handleModal={handleModal} addItem={addItem} text={text} onTextChange={onTextChange} />
            </Modal>
        </View>
    )
}
const styles = StyleSheet.create({
    container: {
        maxHeight:heightPercentageToDP('82.5%'),
    },

    flatview: {
        maxHeight: heightPercentageToDP('70%'),
        height: heightPercentageToDP('77%')
    },

    plus: {
        marginLeft: widthPercentageToDP('78%'),
    }

});

export default NoteTaking;
