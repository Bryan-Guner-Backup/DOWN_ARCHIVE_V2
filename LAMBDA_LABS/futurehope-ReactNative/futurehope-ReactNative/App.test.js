import React from 'react';
import 'react-native';
import { render, fireEvent } from 'react-native-testing-library';

import App from './App';
import Header from './components/Header';
import Home from './components/Home';
import ListItem from './components/ListItem';
import NoteTaking from './components/NoteTaking';
import AddItem from './components/AddItem';
import NoteTakingEdit from './components/NoteTakingEdit';

test('App test against snapshot', () => {
    const tree = render(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders the passed label', () => {
  const { queryByText } = render(<NoteTaking item="Test Item" />);
  expect(queryByText('ASDF')).toBeNull();
});

test('can press the button', () => {
  const onPressMock = jest.fn();
  const dummyPressMock = jest.fn()

  const { getByText } = render(<AddItem title="" onTextChange={dummyPressMock} addItem={onPressMock} text="Testing" />);
  const addButton = getByText(/Add Note/i);

  fireEvent.press(addButton);
  expect(onPressMock).toHaveBeenCalled();
  expect(onPressMock.mock.calls.length).toBe(1);

  fireEvent.press(addButton);
  expect(onPressMock.mock.calls.length).toBe(2);
});
