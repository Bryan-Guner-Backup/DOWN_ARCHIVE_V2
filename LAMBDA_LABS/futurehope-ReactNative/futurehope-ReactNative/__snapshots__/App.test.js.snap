// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ff9800",
        "flexDirection": "row",
        "height": 75,
        "justifyContent": "space-between",
        "paddingTop": 28,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "textAlign": "center",
        }
      }
    >
      School in the Sky
    </Text>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 650,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        style={
          Object {
            "backgroundColor": "#ff9800",
            "borderRadius": 70,
            "color": "white",
            "fontSize": 35,
            "margin": 60,
            "padding": 40,
            "textAlign": "center",
            "width": 300,
          }
        }
      >
        <Text />
          Notes
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        style={
          Object {
            "backgroundColor": "#ff9800",
            "borderRadius": 70,
            "color": "white",
            "fontSize": 35,
            "margin": 60,
            "padding": 40,
            "textAlign": "center",
            "width": 300,
          }
        }
      >
        <Text />
         Calendar
      </Text>
    </View>
  </View>
</View>
`;
